topic-operator/src/test/java/io/strimzi/operator/topic/BatchingTopicControllerTest.java

```diff
@@ -476,4 +480,77 @@ public void replicasChangeShouldCompleteWhenCruiseControlRestarts() {
+
+    @Test
+    public void shouldNotCallGetClusterConfigWhenCustomSaslSet() {
+        var admin = mock(Admin.class);
+        var replicasChangeClient = mock(ReplicasChangeHandler.class);
+
+        var config = TopicOperatorConfig.buildFromMap(Map.of(
+              TopicOperatorConfig.BOOTSTRAP_SERVERS.key(), "localhost:1234",
+              TopicOperatorConfig.NAMESPACE.key(), "some-namespace",
+              TopicOperatorConfig.SASL_ENABLED.key(), "true",
+              TopicOperatorConfig.SKIP_CLUSTER_CONFIG_REVIEW.key(), "true"
+        ));
+
+        new BatchingTopicController(config, Map.of("key", "VALUE"), admin, client, metrics, replicasChangeClient);
+
+        verifyNoInteractions(admin);
+    }
+
+    @Test
+    public void shouldNotCallDescribeClusterWhenSkipClusterConfigReviewIsTrue(KafkaCluster cluster) {
+        admin[0] = Admin.create(Map.of(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, cluster.getBootstrapServers()));
+        Admin adminSpy = Mockito.spy(admin[0]);
+
+        var config = mock(TopicOperatorConfig.class);
+        Mockito.doReturn(NAMESPACE).when(config).namespace();
+        Mockito.doReturn(true).when(config).skipClusterConfigReview();
+        var replicasChangeClient = mock(ReplicasChangeHandler.class);
+
+        controller = new BatchingTopicController(config, Map.of("key", "VALUE"), adminSpy, client, metrics, replicasChangeClient);
+
+        verifyNoInteractions(adminSpy);
+    }
+
+    @Test
+    public void shouldNotUpdatePropertiesNotInTheAlterableProperties(KafkaCluster cluster) throws InterruptedException, ExecutionException {
+        admin[0] = Admin.create(Map.of(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, cluster.getBootstrapServers()));
+
+        admin[0].createTopics(List.of(new NewTopic(MY_TOPIC, 1, (short) 1).configs(Map.of(TopicConfig.COMPRESSION_TYPE_CONFIG, "snappy")))).all().get();
+
+        Admin adminSpy = Mockito.spy(admin[0]);
+
+        var configResources = new ConfigResource(ConfigResource.Type.TOPIC, MY_TOPIC);
+        var describeConfigsResultReal = adminSpy.describeConfigs(Set.of(configResources));
+        var topicConfig = describeConfigsResultReal.values().get(configResources).get();
+
+        var config = mock(TopicOperatorConfig.class);
+        Mockito.doReturn(NAMESPACE).when(config).namespace();
+        Mockito.doReturn(true).when(config).skipClusterConfigReview();
+        Mockito.doReturn("compression.type, max.message.bytes, message.timestamp.difference.max.ms, message.timestamp.type, retention.bytes, retention.ms").when(config).alterableTopicConfig();
+        var replicasChangeClient = mock(ReplicasChangeHandler.class);
+
+        // Setup the KafkaTopic with 1 property change that is not in the alterableTopicConfig() list.
+        Map<String, Object> deepCopyOfTopicConfig = topicConfig.entries().stream()
+              .collect(Collectors.toMap(ConfigEntry::name, ConfigEntry::value));
+        deepCopyOfTopicConfig.put("cleanup.policy", "Compact");
```

Comment:
Why you need to create a deep copy? Can't you simply do:

```sh
...
.withConfig(Map.of(
    TopicConfig.COMPRESSION_TYPE_CONFIG, "snappy",
    TopicConfig.CLEANUP_POLICY_CONFIG, "compact"))
...
```

Reply:
Good point.